src/scr_util_mpi.c

```diff
@@ -352,7 +355,7 @@ int scr_axl(
       scr_err("Failed to add file to AXL transfer handle %d: %s --> %s @ %s:%d",
         id, src_file, dest_file, __FILE__, __LINE__
       );
-      rc = SCR_FAILURE;
+      return SCR_FAILURE;
```

Comment:
I think we need to keep this with a set rather than a return to avoid deadlock below.

Reply:
Not sure I follow - which deadlock are you referring to?  Do we need to still call `AXL_Free_comm()` on failures?

Reply:
The particular deadlock I'm worried about here happens if ``AXL_Add`` fails on one process, but succeeds on other procs.  In that case, that process would now return with this change whereas the other procs will go on to call ``AXL_Dispatch_comm``, but that contains an allreduce, which leads to the deadlock.

On the other hand, the existing code (before your change here) was also broken.  We shouldn't call ``AXL_Dispatch`` if ``AXL_Add`` fails.  I think we need an allreduce inbetween to check that ``AXL_Add`` succeeded everywhere.  If not we need to call ``AXL_Free`` and return everywhere with an error.  I think a second problem in the existing code is that we shouldn't call ``AXL_Wait`` if ``AXL_Dispatch_comm`` returns an error.  There may be some other related problems with the existing code.

One option might be to drop these changes from the PR and open a separate issue to clean all of this up in a separate PR.